hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/SparkDatasetMixin.scala

```diff
@@ -29,9 +29,17 @@ import scala.collection.JavaConverters._
 trait SparkDatasetMixin {
 
   def toDataset(spark: SparkSession, records: java.util.List[HoodieRecord[_]]) = {
-    val avroRecords = records.asScala.map(
-      _.getData
-        .asInstanceOf[GenericRecord]
+    val record1 = records.get(0)
+    val isSerializableIndexedRecord = record1.getData.isInstanceOf[SerializableIndexedRecord]
```

Comment:
how much perf are we saving by doing this instead of checking each record? Also need to handle empty list so we don't get index oob

Reply:
this is just test right. I did benchmark. 
as of now, this is only used while ingesting new data into hudi. and hence does not have any delete handling. we don't need to account for deleted records in this case.

Reply:
ok